--- Makefile.PL.orig	Mon Jul 21 23:50:14 2008
+++ Makefile.PL	Tue Jul 22 00:41:44 2008
@@ -397,7 +397,18 @@
 ),
 );
 
-sub MY::postamble
+package MY;
+use Config;
+
+sub dynamic_lib {
+	my $inherited = shift->SUPER::dynamic_lib(@_);
+	$inherited .= "\tdlltool --input-def Cairo.def --dllname Cairo.dll \\\n";
+	$inherited .= "\t\t--output-lib libCairoPerl.a\n";
+	$inherited .= "\t\$(CP) libCairoPerl.a " . $Config{libpth} . "\n";
+	return $inherited;
+}
+
+sub postamble
 {
 	my $autogen_dir = $MakeHelper::autogen_dir;
 
